### ISSUE TITLE
Future direction?

### ISSUE THREAD

**Issue by tfausak** [tfausak/grawlix#2]
Grawlix is at the proof of concept stage right now. Using a bookmarklet, you can comment on Haskell documentation. It works with Hackage and Stackage. It works in Chrome, Safari, and Firefox. You can authenticate with GitHub. It's not pretty, but all the pieces are there. 

So what's next? @eborden pointed me to thoughtbot's [Carnival](https://carnivalapp.io), which is the same idea except it works on any site instead of only Haskell documentation. (Apparently @pbrisbin had a hand in making that. Hi Pat!) Honestly something like that seems better. Why limit yourself to one site?

I don't think I'll keep plugging away at Grawlix to try and beat thoughbot at the Carnival game. Maybe they can convince Hackage and/or Stackage to include Carnival. Or I could make a Carnival site and drop the script in the DOM with a bookmarklet. Either way, they've got me beat.

Why comment on Haskell documentation at all? The end goal was always *better documentation*, not commenting. I think comments are a great way to improve documentation (see [Real World Haskell](http://book.realworldhaskell.org/read/why-functional-programming-why-haskell.html#id528893) or any PHP documentation) but they're not the only way. 

I think Grawlix could pivot, so to speak, into providing better Haskell documentation. There's a lot to like about the current state of the art, but it's scattered around and hard to make sense of. Hackage is the source of truth; Hoogle and Hayoo can search, but their indexes are small; Stackage has a better index and more reliable documentation; random other sites provide things like reverse dependencies or licenses. Some things are harder than they should be, like getting to the latest version of a package. Lots of good documentation is stored but not indexed, like READMEs on GitHub. Other features are (apparently) missing, like complete dependency trees. 

Pulling all that together under one roof and providing a consistent interface to it would have real value in my eyes. It would change Grawlix from being a bookmarklet that scribbles on other sites into its own site that draws from a bunch of sources and adds its own stuff. 

Anyway, I've been thinking about this since I "announced" Grawlix [on Twitter](https://twitter.com/taylorfausak/status/910149136351207424) last week. It got a lot of interest (16 retweets and 33 favorites, which isn't earth-shattering but it is respectable for #Haskell) but pretty much nobody commented. Obviously engagement for a bookmarklet-based (or even developer console-based) thing is going to be pretty low, but I think only one person made a comment. (Thanks @vincenthz!) 

One obvious way to boost engagement is to make it a real site rather than a bookmarklet. But that would require wrapping some other site's documentation, which would require a bunch of other stuff. So why not turn it into a real site that actually adds value rather than a thin wrapper? That's where I'm at.

TL;DR: [ClojureDocs](https://clojuredocs.org) or [Grimoire](https://www.conj.io), but for Haskell?
